code/datums/spells/shapeshift.dm

```diff
@@ -0,0 +1,70 @@
+/obj/effect/proc_holder/spell/targeted/shapeshift
+	name = "Shapeshift"
+	desc = "Change shapes"
+	clothes_req = 0
+	human_req = 0
+	charge_max = 100
+	range = -1
+	include_user = 1
+	invocation = "RAC'WA NO!"
+	invocation_type = "shout"
+	action_icon_state = "shapeshift"
+
+	var/shapeshift_type = /mob/living/simple_animal/hostile/retaliate/goat
+	var/list/current_shapes= list()
+	var/list/current_casters = list()
+
+/obj/effect/proc_holder/spell/targeted/shapeshift/cast(list/targets)
+	for(var/mob/living/M in targets)
+		if(M in current_shapes)
+			Restore(M)
+		else
+			Shapeshift(M)
+
+/obj/effect/proc_holder/spell/targeted/shapeshift/proc/Shapeshift(mob/living/caster)
+	for(var/mob/living/M in caster)
+		if(M.status_flags & GODMODE)
+			caster << "<span class='warning'>You're already shapeshifted!</span>"
+			return
+
+	var/mob/living/shape = new shapeshift_type(caster.loc)
+	caster.loc = shape
+	caster.status_flags |= GODMODE
+	
+	current_shapes |= shape
+	current_casters |= caster
+	clothes_req = 0
+	human_req = 0
```

Comment:
I left that in case i wanted to make the wizard version need clothes, but decided on just leaving it as robeless spell anyway.